src/Commands/PluginCommand.php

```diff
@@ -0,0 +1,568 @@
+    {
+        if (empty($project_list)) {
+            $message = "Usage: terminus plugin:<install|add>";
+            $message .= " <URL to plugin archive, Git or Packagist project 1>,";
+            $message .= "[URL to plugin archive, Git or Packagist project 2],...";
+            $this->log()->error($message);
+            return false;
+        }
+
+        $projects = explode(',', $project_list);
+        $plugins_dir = $this->getPluginDir();
+        foreach ($projects as $project) {
+            $is_url = (filter_var($project, FILTER_VALIDATE_URL) !== false);
+            if (!$is_url) {
+                if (!$this->isValidPackagistProject($project)) {
+                    $message = "{$project} is not a valid Packagist project.";
+                    $this->log()->error($message);
+                } else {
+                    $path = explode('/', $project);
+                    $plugin = $path[1];
+                    if (is_dir($plugins_dir . $plugin)) {
+                        $message = "{$plugin} plugin already installed.";
+                        $this->log()->notice($message);
+                    } else {
+                        exec("composer create-project --dev -n -d {$plugins_dir} {$project}", $output);
+                        foreach ($output as $message) {
+                            $this->log()->notice($message);
+                        }
+                    }
+                }
+            } else {
+                if ($type = $this->getInstallType($project)) {
+                    switch ($type) {
+                        case 'git':
+                            $parts = parse_url($project);
+                            $path = explode('/', $parts['path']);
+                            $plugin = array_pop($path);
+                            $repository = $parts['scheme'] . '://' . $parts['host'] . implode('/', $path);
+                            if (!$this->isValidGitRepository($repository, $plugin)) {
+                                $message = "{$project} is not a valid plugin Git repository.";
+                                $this->log()->error($message);
+                            } else {
+                                if (is_dir($plugins_dir . $plugin)) {
+                                    $message = "{$plugin} plugin already installed.";
+                                    $this->log()->notice($message);
+                                } else {
+                                    exec("git clone --branch 1.x {$project} {$plugins_dir}{$plugin}", $output);
+                                    foreach ($output as $message) {
+                                        $this->log()->notice($message);
+                                    }
+                                }
+                            }
+                            break;
+
+                        case 'archive':
+                            exec("curl {$project} -L | tar -C {$plugins_dir} -xvz", $output);
```

Comment:
Has anyone confirmed this works on Windows 10?  It probably does not work on Windows 7.  Windows 7 and 8 are receiving maintenance releases only, but still account for a very large portion of the general Windows population. Terminus users are probably more heavily skewed towards Mac / Linux / Win10 than the general population, though.